File: src/main/java/xyz/phanta/cbtweaker/buffer/ingredient/IngredientMatcher.java
package xyz.phanta.cbtweaker.buffer.ingredient;

import xyz.phanta.cbtweaker.integration.jei.ingredient.JeiIngredientProvider;

import java.util.function.Supplier;

public interface IngredientMatcher<A, JA> extends JeiIngredientProvider<JA> {

    default boolean consumeInitial(Supplier<A> acc, float consumeFactor) {
        return true;
    }

    default boolean consumePeriodic(Supplier<A> acc, float consumeFactor) {
        return true;
    }

}

File: src/main/kotlin/st/evening/mc/cbtweaker/buffer/ingredient/IngredientMatcher.kt
package st.evening.mc.cbtweaker.buffer.ingredient

import st.evening.mc.cbtweaker.compat.jei.ingredient.JeiIngredientProvider

interface IngredientMatcher<A, JA> : JeiIngredientProvider<JA> {
    fun consumeInitial(acc: Lazy<A>, consumeFactor: Float, determMode: Boolean): Boolean = true

    fun consumePeriodic(acc: Lazy<A>, consumeFactor: Float): Boolean = true
}
